src/Security/Authentication/Core/src/SystemClock.cs

```diff
@@ -6,8 +6,26 @@ namespace Microsoft.AspNetCore.Authentication;
 /// <summary>
 /// Provides access to the normal system clock with precision in seconds.
 /// </summary>
+[Obsolete("Use TimeProvider instead.")]
 public class SystemClock : ISystemClock
 {
+    private readonly TimeProvider _time;
+
+    /// <summary>
+    /// Creates a new SystemClock that reads the current system time.
+    /// </summary>
+    public SystemClock() : this(TimeProvider.System) { }
+
+    /// <summary>
+    /// Creates a new SystemClock that reads from the given TimeProvider.
+    /// </summary>
+    /// <param name="time">The authoritative <see cref="TimeProvider"/>.</param>
+    public SystemClock(TimeProvider time)
```

Comment:
We didn't discuss this in API review. I also don't think it's necessary. We should encourage any new code to use `TimeProvider` directly and not reference `SystemClock` at all. It's very easy to create your own `ISystemClock` implementation based on `TimeProvider` if you really want to.

Reply:
This was needed in the auth handler and the validator to populate the old ISystemClock Clock property with the new TimeProvider so they'd return the same result. Since I needed it at least twice it seemed useful to include on the existing type.

Reply:
I'm not sure us needing it in two places is indicative of many others needing it. Library authors, maybe. Apps authors, almost certainly not. And it's super easy to implement yourself. Either way, it's worth quickly mentioning in the API review issue and via email/meeting to get eyes on it since it is a public API change.